### PR TITLE
Refactor OBO Graph JSON parser

### PR DESCRIPTION
**Pull request by cthoyt**
This PR does the following:

1. massively reduces indentation
2. renames variables in a nice way
3. refactors out shared code into a helper function
4. It also changes so it doesn't add an empty string for subject or object label, and instead doesn't add anything

### CODE REVIEW

File: src/sssom/parsers.py [hunk 1]
```diff
@@ -68,6 +68,8 @@
 
 logging = _logging.getLogger(__name__)
 
+DBXREF_URI = "http://www.geneontology.org/formats/oboInOwl#hasDbXref"
```
Comment (matentzn) [inline src/sssom/parsers.py:71]: This should be imported from constants.py (probably its already there)

File: src/sssom/parsers.py [hunk 2]
```diff
@@ -566,118 +568,104 @@ def from_obographs(
...
+                continue
+            mdict = _make_mdict(edge["sub"], predicate_uri, edge["obj"], converter, labels)
+            _add_valid_mapping_to_list(mdict, mlist)
+
+        if OWL_EQUIV_CLASS_URI in mapping_predicates:
+            for equivalents in graph.get("equivalentNodesSets", []):
+                node_ids = equivalents.get("nodeIds")
+                if not node_ids:
+                    continue
+                for subject_uri, object_uri in itt.product(node_ids, repeat=2):
```
Comment (matentzn) [inline src/sssom/parsers.py:618]: What is itt?

**Review by matentzn — COMMENTED** [review]
Beautiful refactoring of my terrible code 🗡️ 

I hope the tests are strong enough to ensure nothing is lost. 

@hrshdhgd can you check that when using this branch, mondo.sssom.tsv is still generated correctly (i.e. no rows are lost compared to normal sssom-py)

File: src/sssom/parsers.py (hunk 2 above)
Comment (cthoyt) [reply src/sssom/parsers.py:618]: itertools. Importing as `itt` is very common

File: src/sssom/parsers.py (hunk 1 above)
Comment (cthoyt) [reply src/sssom/parsers.py]: Done in 6c09ed3

File: src/sssom/parsers.py (hunk 2 above)
Comment (matentzn) [reply src/sssom/parsers.py:619]: Thanks

**Comment by hrshdhgd** [comment]
[Just ran it](https://github.com/monarch-initiative/mondo/pull/6872/commits/2bfcad003e18bb12203c89d7ccc3e8008946d600) and the differences aren't major.

**Review by matentzn — APPROVED** [review]
Thank you @cthoyt for this PR, and @hrshdhgd for checking!

**Pull request merged**